File: Modules/Common/Techs/SIE.Web.Tech/Stations/Commands/StationEquipmentSetMasterCommand.js
SIE.defineCommand('SIE.Web.Tech.Stations.Commands.StationEquipmentSetMasterCommand', {
    meta: { text: "设置为主设备", group: "edit", iconCls: "icon-ListConfig icon-blue" },

    canExecute: function (view) {
        if (view.getSelection().length === 1 && view.getCurrent() !== null) {
            var curr = view.getCurrent();
            return curr.isNew() != true && curr.getData().IsMaster === false;
        }
        return false;
    },
    execute: function (view, source) {
        var entity = view.getCurrent();
        if (entity) {
            entity = entity.data;
            var msg = Ext.String.format('是否将{0}设置为主设备？'.t(), entity.Name);
            SIE.Msg.askQuestion(msg, function () {
                view.execute({
                    data: entity,
                    success: function (res) {
                        view.loadData();
                    }
                }, view);
            });
        }
    }
});

File: Modules/Common/Techs/SIE.Web.Tech/Stations/Commands/StationEquipmentSetMasterCommand.cs
using SIE.Tech.Stations;
using SIE.Web.Command;
using System;

namespace SIE.Web.Tech.Stations.Commands
{
    /// <summary>
    /// 设置主设备
    /// </summary>
    public class StationEquipmentSetMasterCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var entity = args.Data.ToJsonObject<StationEquipment>();
            if (null == entity)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(entity)));
            }
            RT.Service.Resolve<StationController>().SetEquipmentMaster(entity);
            return true;
        }
    }
}
